File: model/src/main/java/io/quartic/geojson/MultiPoint.java
package io.quartic.geojson;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import io.quartic.common.SweetStyle;
import org.immutables.value.Value;

import java.util.List;

@SweetStyle
@Value.Immutable
@JsonSerialize(as = MultiPointImpl.class)
@JsonDeserialize(as = MultiPointImpl.class)
public interface MultiPoint extends Geometry {
    List<List<Double>> coordinates();


    @Override
    default <T> T accept(GeometryVisitor<T> geometryVisitor) {
        return geometryVisitor.visit(this);
    }
}

File: common-core/src/main/kotlin/io/quartic/common/geojson/MultiPoint.kt
package io.quartic.common.geojson

import com.fasterxml.jackson.annotation.JsonTypeName

@JsonTypeName("MultiPoint")
data class MultiPoint @JvmOverloads constructor(
        val coordinates: List<List<Double>>,
        override val crs: Map<String, Any>? = null,
        override val bbox: List<Double>? = null
) : Geometry